### DISCUSSION TITLE
[ZK Stack] - How to run multiple chains under a ecosystem with same ledger?

### DISCUSSION THREAD

**Discussion by sachanayush47** [zkSync-Community-Hub/zksync-developers#772] [ZK Stack] [answered]
### Module Affected

- [ ] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [ ] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [X] Other (please specify below)

### Rust Version

1.82.0

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

How can we operate multiple zkSync Hyperchains within a unified ecosystem that maintains a shared ledger across all chains? For instance, consider two chains within this ecosystem, CHAIN_1 and CHAIN_2. If a particular address holds a balance of 15 ETH on CHAIN_1, the ecosystem should synchronize balances such that CHAIN_2 also reflects the same 15 ETH balance for that address.

### Expected Behavior

NA

### Current Behavior

NA

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
Currently, the ZKsync documentation and discussions do not provide specific details on operating multiple ZKsync Hyperchains within a unified ecosystem that maintains a shared ledger across all chains. This type of setup would involve complex synchronization of state and balances across different chains, which is not covered in the available resources.

For more detailed guidance and potential updates on this topic, I recommend visiting the ZKsync developers discussions on Github at [ZKsync developers discussions](https://github.com/zkSync-Community-Hub/zksync-developers/discussions). Here, you can engage with other developers and possibly find more insights or share your own developments regarding multi-chain ecosystems within ZKsync.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by popzxc** [comment, marked as answer]
Hello! Right now, it's not possible to do such a setup. With [interop](https://github.com/matter-labs/zksync-era/blob/main/docs/specs/zk_chains/interop.md), it would be possible to enable seamless interaction between chains, although they won't be "fully synchronized", e.g. the same address may have different balances on Chain 1 and Chain 2. Ability to run "synchronized" chains isn't currently planned.
